substrate/frame/revive/src/gas.rs

```diff
@@ -168,27 +168,14 @@ impl<T: Config> GasMeter<T> {
 		}
 	}
 
-	/// Create a new gas meter by removing gas from the current meter.
+	/// Create a new gas meter for a nested call by removing gas from the current meter.
 	///
-	/// # Note
-	///
-	/// Passing `0` as amount is interpreted as "all remaining gas".
+	/// Per [EIP-150](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-150.md), this is
+	/// capped at 63/64ths of the caller's gas limit.
 	pub fn nested(&mut self, amount: Weight) -> Self {
-		let amount = Weight::from_parts(
-			if amount.ref_time().is_zero() {
-				self.gas_left().ref_time()
-			} else {
-				amount.ref_time()
-			},
-			if amount.proof_size().is_zero() {
-				self.gas_left().proof_size()
-			} else {
-				amount.proof_size()
-			},
-		)
-		.min(self.gas_left);
-		self.gas_left -= amount;
-		GasMeter::new(amount)
+		let amt = amount.min(self.gas_left - self.gas_left / 64);
+		self.gas_left -= amt;
```

Comment:
Please use `amount` and not `amt`. No need to introduce a less readable identifier when we can just shadow.